bin/ci/verify-pr-build.sh

```diff
@@ -0,0 +1,70 @@
+#!/bin/bash
```

Comment:
After this PR merges, we want to start adding functional tests to plans. Those tests will require a package build. If we're then doing a pkg build twice in CI we can maybe:
 * accept the duplicate pkg builds, at least short term.
 * have this script `buildkite artifact upload` what it has built and have our functional tests `artifact download` the package.
 * exclude plans we add functional tests to from this build verification step - potentially this script then gets retired.

Reply:
I think we'll have to accept duplicate package builds.  Running with DO_CHECK may alter the resulting artifact. To confirm we'd have to verify every package that has a `do_check` and reverify every version bump as the upstream may have made modifications. Given that I think it's cheaper to accept the extra build.  

What we should do is, as we expand our test coverage, make this step exit(pass) quickly if there is no `do_check` defined and it has tests.  I think we always want to execute this step if there is a `do_check` present, as upstream often provides extensive testing that we should be taking advantage of.